File: apps/api/app/schemas/feedback.py
"""
Request/response schemas for answer feedback.
"""
from typing import Any, Dict, List, Optional

from pydantic import BaseModel, Field


class FeedbackSubmitRequest(BaseModel):
    session_id: str
    message_id: str
    feedback_type: str = Field(..., pattern="^(up|down)$")
    comment: Optional[str] = Field(None, max_length=2000)
    question_text: Optional[str] = None
    answer_text: Optional[str] = None
    sources: Optional[List[Dict[str, Any]]] = None
    debug: Optional[Dict[str, Any]] = None


class FeedbackSubmitResponse(BaseModel):
    status: str = "ok"
    feedback_id: str

File: apps/web/src/lib/types/feedback.ts
/**
 * Frontend types for answer feedback (aligned with backend).
 */

export type FeedbackType = "up" | "down";

export interface FeedbackSubmitRequest {
  session_id: string;
  message_id: string;
  feedback_type: FeedbackType;
  comment?: string;
  question_text?: string;
  answer_text?: string;
  sources?: unknown[];
  debug?: Record<string, unknown>;
}

export interface FeedbackSubmitResponse {
  status: string;
  feedback_id: string;
}
